### PR TITLE
[spark] Add show tags sql syntax

### PR DESCRIPTION
**Pull request by askwang**
### Purpose

Add show tags sql syntax like show partitions. 
```
show tags table_identifier
```

### CODE REVIEW

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/ShowTagsExec.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    val table = catalog.loadTable(ident)
+    assert(table.isInstanceOf[SparkTable])
+
+    var tags: Seq[InternalRow] = Nil
+    table.asInstanceOf[SparkTable].getTable match {
+      case paimonTable: FileStoreTable =>
+        val tagNames = paimonTable.tagManager().allTagNames()
+        tags = tagNames.asScala.toList.sorted.map(t => InternalRow(UTF8String.fromString(t)))
+      case t =>
+        throw new UnsupportedOperationException(s"Can not show tags for non-paimon table: $t")
```
Comment (wwj6591812) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/ShowTagsExec.scala:45]: Can not show tags for non-paimon FileStoreTable

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/PaimonStrategy.scala [hunk 2]
```diff
@@ -18,16 +18,25 @@
...
 
-case class PaimonStrategy(spark: SparkSession) extends Strategy with PredicateHelper {
+import scala.collection.JavaConverters._
+
+case class PaimonStrategy(spark: SparkSession)
+  extends Strategy
+  with PredicateHelper
+  with PaimonLookupCatalog {
+
+  protected lazy val catalogManager = spark.sessionState.catalogManager
```
Comment (wwj6591812) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/PaimonStrategy.scala:39]: 1、Why protected?
2、In my idea, it says : "Type annotation required for protected member".

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/PaimonStrategy.scala [hunk 3]
```diff
@@ -37,6 +46,8 @@ case class PaimonStrategy(spark: SparkSession) extends Strategy with PredicateHe
     case c @ PaimonCallCommand(procedure, args) =>
       val input = buildInternalRow(args)
       PaimonCallExec(c.output, procedure, input) :: Nil
```
Comment (wwj6591812) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/PaimonStrategy.scala:48]: Add a blank line?

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/PaimonStrategy.scala [hunk 4]
```diff
@@ -48,4 +59,16 @@ case class PaimonStrategy(spark: SparkSession) extends Strategy with PredicateHe
     new GenericInternalRow(values)
   }
 
+  private object PaimonCatalogAndIdentifier {
```
Comment (wwj6591812) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/PaimonStrategy.scala:62]: Why not paimonCatalogAndIdentifier?

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/PaimonStrategy.scala [hunk 5]
```diff
@@ -48,4 +59,16 @@ case class PaimonStrategy(spark: SparkSession) extends Strategy with PredicateHe
...
+    def unapply(identifier: Seq[String]): Option[(TableCatalog, Identifier)] = {
+      val catalogAndIdentifer =
```
Comment (wwj6591812) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/PaimonStrategy.scala:64]: catalogAndIdentifier

**Comment by wwj6591812** [comment]
Could you add this syntax in doc https://paimon.apache.org/docs/master/spark/sql-ddl/ ?

**Comment by askwang** [comment]
> Could you add this syntax in doc https://paimon.apache.org/docs/master/spark/sql-ddl/ ?

doc added. and addressed comments, thanks @wwj6591812

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/PaimonStrategy.scala (hunk 2 above)
Comment (askwang) [reply paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/PaimonStrategy.scala:39]: Protected type is inherited from the parent interface `PaimonLookupCatalog`, cannot be changed.

Type annotation means that you need to specify the variable type like `val catalogManager : CatalogManager`, but class CatalogManager is inaccessible from here due to the limitation of package name.

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/PaimonStrategy.scala (hunk 4 above)
Comment (askwang) [reply paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/execution/PaimonStrategy.scala:63]: It's an extractor object  with an unapply method in Scala, not a member method of class PaimonStrategy.

**Comment by wwj6591812** [comment]
+1

**Pull request merged**